airflow/www/views.py

```diff
@@ -869,7 +869,13 @@ def log(self, session=None):
             models.TaskInstance.task_id == task_id,
             models.TaskInstance.execution_date == dttm).first()
 
-        logs = [''] * (ti.next_try_number - 1 if ti is not None else 0)
+        num_logs = 0
+        if ti is not None:
+            num_logs = ti.next_try_number - 1
```

Comment:
I never was happy with this whole next_try_number hack - I should find time to have one row per try instead
of a row per TI that gets altered.

Reply:
@ashb I'd like that as well! Is there an issue for that work?

Reply:
I don't think so. We'd need to work out how to do the migration so that viewing old tasks still works.

If you create a Jira ticket we can discuss some of the details on there.

Reply:
Another thing I would like to remove: "clear"ing tasks as it deletes history :) Instead I would like a "retry" button in the UI and ending up with "Attempt 3 of 2" sort of thing.

The other advantage of having each try/attempt being a row is that we could then store the URL of the log against that row, to have old logs still be readable even after a future change to Airflow's logging pattern. (Both 1.9 and 1.10 made logs from previous versions no-longer-readable in the UI.)